Document SBT mint and registry opt-in mechanism (#19)
near-ndc/i-am-human: contracts/registry/README.md

> # SBT opt-in

Reviewer: First level header is usually a title, and there should be only one first level header.
Below the header let's start with a summary, and link to the registry documentation of NEP-393.

Author: DONE

> Every SBT smart contract that wants to relay the state changes on the registry must opt-in into the registry. Different registries may implement different mechanisms for opt-in. In the current implementation the only way to opt-in as an issuer is by calling `admin_add_sbt_issuer`. The transaction must be signed by the admin.

Reviewer: ```suggestion
Every SBT smart contract must opt-in to a registry, or implement registry functionality by it's own. Different registries may implement different mechanisms for opt-in. 

This implementation requires an admin address (could be a DAO) to add an issuer to the registry, and as a consequence allow the issuer to use SBT registry methods.
```

# SBT Registry

The Registry smart contract is a balance book for all associated SBT tokens. The registry enables atomic `soul_transfers` and provides an efficient way to ban a smart contract (issuer). For more details check the [nep-393](https://github.com/near/NEPs/pull/393).

## SBT opt-in

Every SBT smart contract must opt-in to a registry, or implement registry functionality by it's own. Different registries may implement different mechanisms for opt-in. 

> This implementation requires an admin address (could be a DAO) to add an issuer to the registry, and as a consequence allow the issuer to use SBT registry methods.

Reviewer: ```suggestion
This implementation requires an admin account (could be a DAO) to add an issuer to the registry, and as a consequence allow the issuer to use SBT registry methods.
```

## SBT mint

The minting process is a procedure where we asign a new token to the provided reciver and keep track of it in the registry. The `sbt_mint` method must be called by a issuer that is opted-in. Additionaly:

- each `TokenMetadata` provided must have a non zero `class`,

> - enough `Near` to cover the registry storage cost must be provided.

Reviewer: ```suggestion
- enough `Near` must be attached to cover the registry storage cost must be provided.
```

> The method will emit the `Mint` event when succesful. There might be a case when the token vector provided is too long, and the gas is not enought to cover the minting process, then it will panic with `out of gas`. 

Reviewer: ```suggestion
The method will emit the [`Mint`](https://github.com/alpha-fi/i-am-human/blob/master/contracts/sbt/src/events.rs#L69) event when succesful. There might be a case when the token vector provided is too long, and the gas is not enought to cover the minting process, then it will panic with `out of gas`. 
```
